### PR TITLE
Feat: :rocket: use external-secret in multiple namespaces

### PR DESCRIPTION
**Pull request by h1manshu98**
## What
- Updated `istio-ingress` addon to use Internet-facing and Internal ingresses.
- Added feature to use externalSecret in other namespaces.
- To do so User need to create an TrustRelationship policy where user will pass information of other namespace.
- Then in their application's service-account.yaml, user have to pass IAMRole annotation as below -
```yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  name: 
  annotations:
    eks.amazonaws.com/role-arn: "arn:aws:iam::12345678912:role/external-secrets-CLUSTER_NAME"
```
## Why
- Previously we were only able to use external-secret in the namespace where this addon is deployed [kube-system].
- To use this external-secret in different namespace-X we had to create a ServiceAccount in namespace-X with AWSSecretManager permission.

### CODE REVIEW

**Review by nileshgadgi — APPROVED** [review]
LGTM!

File: _examples/complete/providers.tf [hunk 1]
```diff
@@ -37,4 +37,6 @@ data "aws_eks_cluster" "eks_cluster" {
   name       = module.eks.cluster_name
   depends_on = [module.eks.cluster_id]
 }
-data "aws_availability_zones" "available" {}
\ No newline at end of file
+data "aws_availability_zones" "available" {}
+data "aws_region" "current" {}
```
[the remark this answers is not in the document]
Comment (h1manshu98) [reply _examples/complete/providers.tf]: done

File: addons/external-secrets/variables.tf [hunk 2]
```diff
@@ -35,16 +35,14 @@ variable "addon_context" {
...
-  })
-}
-
 variable "external_secrets_extra_configs" {
   description = "Override attributes of helm_release terraform resource"
   type        = any
   default     = {}
+}
+
+variable "irsa_assume_role_policy" {
```
[the remark this answers is not in the document]
Comment (h1manshu98) [reply addons/external-secrets/variables.tf]: done

**Review by themaniskshah — APPROVED** [review]
LGTM!

**Pull request merged**